spec/compiler/codegen/primitives_spec.cr

```diff
@@ -237,4 +237,19 @@ describe "Code gen: primitives" do
       Foo.new.crystal_type_id == Foo.crystal_instance_type_id
       )).to_b.should be_true
   end
+
+  it "uses llvm's va_arg instruction" do
+    mod = codegen(%(
+      struct VaList
+        @[Primitive(:va_arg)]
+        def next(type)
+        end
+      end
+
+      list = VaList.new
+      list.next(Int32)
+      ))
+    str = mod.to_s
+    str.should contain("va_arg %VaList* %list")
```

Comment:
Is there perhaps an executable test which you can do to make sure the whole `VaList` aparatus is working?

for example, pass multiple args into a function and make sure they're accessed correctly

Reply:
Should I do it like the example code above? It'd have to generate C code, link the crystal program with it and run it.

Reply:
Why would you have to generate C code? You can call var-arged functions from crystal.

Reply:
Nope, crystal can't call vararg `fun` functions, it just spouts out "wrong number of arguments" when doing so

Reply:
oh, well this is fine for now then.

Would be nice if crystal would support that in the future.

Reply:
I'm pretty sure it was working...

Reply:
@Sija you can call vararg `fun`s in `lib`s, but not `fun`'s with their bodies written in crystal